compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -545,17 +545,23 @@ fn sanity_check_found_hidden_type<'tcx>(
 /// }
 /// fn func<'a>(x: &'a ()) -> impl Id<Assoc = impl Sized + 'a> { x }
 /// // desugared to
+/// fn func<'a>(x: &'a () -> Outer<'a, Assoc = Inner<'a>> {
+///     // Note that in contrast to other nested items, RPIT type aliases can
+///     // access their parents' generics.
 ///
-/// // hidden type is `&'bDup ()`
-/// // During wfcheck the hidden type of `Inner` is `&'a ()`, but
-/// // `typeof(Inner<'b, 'bDup>) = &'bDup ()`.
-/// // So we walk the signature of `func` to find the use of `Inner<'static, 'a>`
-/// // and then use that to replace the lifetimes in the hidden type, obtaining
-/// // `&'a ()`.
-/// type Outer<'b, 'bDup> = impl Id<Assoc = Inner<'b, 'bDup>>;
-/// // hidden type is `&'cDup ()`
-/// type Inner<'c, 'cDup> = impl Sized + 'cDup;
-/// fn func<'a>(x: &'a () -> Outer<'static, 'a> { x }
+///     // hidden type is `&'aDupOuter ()`
```

Comment:
This feels wrong, if the hidden type of `Outer` is `&'aDupOuter ()`, shouldn't wf check of `Outer` define the hidden type of `Inner` to be `<&'aDupOuter () as Id>::Assoc` which is `&'aDupOuter ()`, so this should worK?

Reply:
No, because we figure out the hidden type of `Inner` anew during wf check of `Outer`, we don't get it from `type_of`. We get the hidden type of `Outer`, which is `&'a ()`, so the id projection is also `&'a ()`